lib/AST/NameLookup.cpp

```diff
@@ -2758,6 +2758,21 @@ void FindLocalVal::checkPattern(const Pattern *Pat, DeclVisibilityKind Reason) {
   
 void FindLocalVal::checkParameterList(const ParameterList *params) {
   for (auto param : *params) {
+    if (param->hasAttachedPropertyWrapper()) {
+      // FIXME: This shouldn't be needed, but only this set the interface type
+      // of the auxiliary variables.
```

Comment:
The reason why `PropertyWrapperAuxiliaryVariablesRequest` does not set the interface type of auxiliary variables is because is because for closures, the interface types are inferred and set by the constraint system. There are more details and test cases that rely on this behavior in https://github.com/apple/swift/pull/36521